### PR TITLE
maxusers: Update TODOs

### LINKED ISSUE #310: max users needs to be a config setting or project wide const
We need to get around to making this a config setting at some point.  Note that inaddition to the 10000 magic number appearing various config files for address derivation, it's also set as a package-level var in the `controller` package: https://github.com/decred/dcrstakepool/blob/f6b21eee683a50f109e367625e540b42c7ffe508/controllers/main.go#L46

_Originally posted by @chappjc in https://github.com/decred/dcrstakepool/pull/308_

### PR DESCRIPTION
**Pull request by JoeGruffins**
~-make max users a configurable variable~
~-default 10000~
~-stakepoold start with default 10000~
~-when starting dcrstakepool grpc commands are used to set the max on all servers and re-derive pool fees if there was a change~

~ahh, meant to do the draft thing, adding WIP to title~

This pr took a different direction mid-stream.  I now think that maxusers is an artificial ceiling and it should be removed. If my understanding is incorrect, please comment.

### CODE REVIEW

File: backend/stakepoold/rpc/api.proto [hunk 1]
```diff
@@ -169,3 +169,8 @@ message GetStakeInfoResponse {
 	double ProportionMissed = 15;
 	uint32 Expired = 16;
 }
+
+message SetMaxUsersRequest {
+	uint32 MaxUsers = 1;
+}
+message Empty{}
```
Comment (JoeGruffins) [inline backend/stakepoold/rpc/api.proto]: @jholdstock 
What do you think about this? If we don't need a value, this can be reused, but it is different from the other code. I can change it to SetMaxUsersResponse.

File: sample-dcrstakepool.conf [hunk 2]
```diff
@@ -143,3 +127,22 @@ designation=YourVSP
...
+
+; Whether to set the secure flag on cookies.  If you have SSL/TLS setup then
+; you should change this to true.
+;cookiesecure=true
+
+; Path to the root folder/directory which contains CSS/fonts/images/javascript.
+;publicpath=public
+
+; Path to the root folder/directory which contains the HTML templates.
+;templatepath=views
```
Comment (JoeGruffins) [inline sample-dcrstakepool.conf]: I was going to rearrange this, but maybe should do that in another pr

**Comment by JoeGruffins** [comment]
I think this is finished. It will change a bit after #463 if that goes through first.

File: backend/stakepoold/rpc/api.proto (hunk 1 above)
Comment (jholdstock) [reply backend/stakepoold/rpc/api.proto]: I would suggest using a dedicated `SetMaxUsersResponse`, even if it includes no fields right now. That will enable adding fields to the response in the future without incrementing the API major version number.

Comment (JoeGruffins) [reply backend/stakepoold/rpc/api.proto]: Hadn't even thought about that. ok.

**Comment by JoeGruffins** [comment]
Will rebase after release.

**Comment by JoeGruffins** [comment]
@jholdstock Let me know your thoughts:
Q. What is max users there for?
My A. To decide the number of fee addresses that should be derived.
Q. Should max users be settable?
A. Yes. As dcrstakepool is now, a pool that is approaching max users must be able to increase this to allow more users.
Q. Should it be set in one place?
A. No preference from me.  But I believe you think this could lead to bugs later on. In which case the appropriate path to take is add maxusers to the stakepoold conf and check that they are all the same on startup. If not let the operator change his/her conf.
Q. What's the future of max users?
A. Lets remove it.  I believe there was talk yesterday on setting all of the fee addresses inside dcrwallet. It shouldn't be done there.  Having a limit on users because of the feeaddresses (if that is the reason for this config setting) is silly.

**Comment by JoeGruffins** [comment]
This pr won't work. stakepoolcoldextkey must also be set on wallet start-up and there's no way to query/set the value afterwards. As I think this entire concept will be changed in the end it's probably not worth making this value settable through rpc commands. As for this pr and the issue, either we should make it settable also in stakepoold conf or make it a constant for now. currently max users can't exceed 10000, even if set above, so making it a constant should not have an affect on any current pools.

**Comment by JoeGruffins** [comment]
If there's no good way to check that dcrwallet.conf value, I think making this settable is a bad idea. I'm going to make maxusers a constant unless there is an objection.

**Comment by JoeGruffins** [comment]
Ok. I think #310 should be closed in order to progress in the direction of  #504.
